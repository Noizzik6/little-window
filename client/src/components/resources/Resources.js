import React from 'react';
import styled from 'styled-components';

const Botresources = styled.div`
  float: left;
  margin-left: 5%;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
`;

const Singleresource = styled.a`
  padding: 0.5rem;
  margin: 5px;
  border: 2px #b0b0b0 solid;
  border-radius: 15px;
`;

const Resources = (props) => {
  if (!props.resources) {
    return null;
  }
  return (
    <Botresources>
      {props.resources.map((resource, index) => (
        <Singleresource href={resource.href} key={index} target="__blank">
          {resource.text}
        </Singleresource>
        ))}
    </Botresources>
  );
};

export default Resources;
