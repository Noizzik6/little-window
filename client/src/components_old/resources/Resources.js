import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { viewLang } from './Languages';

const Botresources = styled.div`
  float: left;
  margin-left: 5%;
  margin-right: 5%;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  a:visited {
    color: blue;
  }
`;

const Singleresource = styled.p`
  padding: 0.5rem;
  margin: 5px;
  border: 2px #b0b0b0 solid;
  border-radius: 15px;
`;

const Singlelink = styled.a`
  margin-left: 10px;
`;

// Resources renders the resources from the props passed down, from the
// text and href properties within the array of resources.
const Resources = (props) => {
  const { resources, lang } = props;

  if (!resources) {
    return null;
  }
  return (
    <Botresources>
      {resources.map((resource) => (
        <div key={resource.text}>
          <Singleresource>
            {resource.text}
            <Singlelink href={resource.href} target="__blank">{viewLang(lang)}</Singlelink>
          </Singleresource>
        </div>
      ))}
    </Botresources>
  );
};

Resources.propTypes = {
  resources: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string.isRequired, href: PropTypes.string.isRequired,
  })),
  lang: PropTypes.string.isRequired,
};

Resources.defaultProps = {
  resources: [],
};

export default Resources;
