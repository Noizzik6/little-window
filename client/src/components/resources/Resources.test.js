/* eslint-env browser */

import React from 'react';
import ReactDOM from 'react-dom';
import Resources from './Resources';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Resources />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Renders correctly with one resource', () => {
  const testResources = [{ text: 'Test 1', href: 'www.chayn.co' }]
  const resourceTest = renderer.create(
    <Resources resources={testResources} />
  ).toJSON();

  expect(resourceTest).toMatchSnapshot();
});

test('Renders correctly with two resources', () => {
  const testResources = [{ text: 'Test 1', href: 'www.chayn.co' },  { text: 'Test 1', href: 'www.chayn.co' }]
  const resourceTest = renderer.create(
    <Resources resources={testResources} />
  ).toJSON();

  expect(resourceTest).toMatchSnapshot();
});
