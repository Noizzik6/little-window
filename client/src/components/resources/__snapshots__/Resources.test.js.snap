// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with one resource 1`] = `
<div
  className="sc-bdVaJa kHsrTA"
>
  <div>
    <p
      className="sc-bwzfXH kGGnuj"
    >
      Test 1
      <a
        className="sc-htpNat gPKckf"
        href="www.chayn.co"
        target="__blank"
      />
    </p>
  </div>
</div>
`;

exports[`Renders correctly with two resources 1`] = `
<div
  className="sc-bdVaJa kHsrTA"
>
  <div>
    <p
      className="sc-bwzfXH kGGnuj"
    >
      Test 1
      <a
        className="sc-htpNat gPKckf"
        href="www.chayn.co"
        target="__blank"
      />
    </p>
  </div>
  <div>
    <p
      className="sc-bwzfXH kGGnuj"
    >
      Test 1
      <a
        className="sc-htpNat gPKckf"
        href="www.chayn.co"
        target="__blank"
      />
    </p>
  </div>
</div>
`;
